storage/s3.go

```diff
@@ -382,6 +386,175 @@ func (s *S3) Get(
 	})
 }
 
+type SelectRecord struct {
+	Record json.RawMessage
+	Err    error
+}
+
+func (s *S3) selectStream(resp *s3.SelectObjectContentOutput, resultch chan<- *SelectRecord) {
+	defer resp.EventStream.Close()
+
+	reader, writer := io.Pipe()
+	defer reader.Close()
+	go func() {
+		defer writer.Close()
+		// TODO: wire ctx up to eventStream.StreamCloser.Close
+		for event := range resp.EventStream.Reader.Events() {
+			switch e := event.(type) {
+			case *s3.RecordsEvent:
+				// TODO: how should we handle errors here?!
+				writer.Write(e.Payload)
+			// How the heck are these different?
+			case *s3.StatsEvent:
+			case *s3.ProgressEvent:
```

Comment:
I think it'd be cool to try and output progress to stderr, but I didn't get to that in this PR.

Reply:
we have an open ticket for this #51.